.github/workflows/dev.yml

```diff
@@ -22,10 +32,134 @@ on:
 
 jobs:
   build:
-    uses: axonivy-market/github-workflows/.github/workflows/ci.yml@v6
-    with:
-      mvnArgs: '"-Divy.engine.download.url=https://dev.axonivy.com/permalink/${{ inputs.ivyVersion }}/axonivy-engine.zip" "-Divy.engine.version=(12.0.0,]" "-Dproject.build.plugin.version=${{ inputs.ivyPluginVersion }}" "-Dtester.version=${{ inputs.testerVersion }}" ${{ inputs.mvnArgs }}'
-      javaVersion: 21
-      mvnVersion: 3.9.8
-    secrets:
-      mvnArgs: '${{ secrets.mvnArgs }}'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup Java JDK
+      uses: actions/setup-java@v4
+      with:
+        java-version: ${{ inputs.javaVersion }}
+        distribution: temurin
+
+    - name: Setup Maven
+      uses: stCarolas/setup-maven@v5
+      with:
+        maven-version: ${{ inputs.mvnVersion }}
+
+    - name: Convert to lastest Ivy version
+      run: |
+        export IVY_JAVA_HOME=$JAVA_HOME_21_X64
+        engineUrl="https://dev.axonivy.com/permalink/${{ inputs.ivyVersion }}/axonivy-engine.zip"
+        echo "workDir=$(pwd)" >> $GITHUB_ENV
+        DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+        if [ -z "$workDir" ]; then
+          workDir=$(mktemp -d -t projectConvertXXX)
+        fi
+
+        if [ -z "$gitDir" ]; then
+          gitDir="$DIR/repos"
+          echo $(mkdir -v -p $DIR/repos)
+        fi
+
+        if ! [ -d "${workDir}/engine" ]; then
+          echo "Downloading engine from ${engineUrl}"
+          (cd "$workDir" && curl --progress-bar -L -O "${engineUrl}") 
+          zipped=$(find "${workDir}" -maxdepth 1 -name "*.zip")
+          unzip -qq "${zipped}" -d "${workDir}/engine"
+          rm "${zipped}"
+        fi
+
+        gitDir=$(pwd)
+        gitName=$(basename ${gitDir})
+        echo "Searching projects in ${gitDir}"
+        projects=()
+        for ivyPref in $(find ${gitDir} -name "ch.ivyteam.ivy.designer.prefs"); do
+          project=$(dirname $(dirname $ivyPref))
+          if ! [ -f "${project}/pom.xml" ]; then
+            continue # prefs file not in natural project structure
+          fi
+          projects+=("${project}")
+        done
+
+        if [ ${#projects[@]} -gt 0 ]; then
+          echo "Collected projects: ${projects[@]}"
+          ${workDir}/engine/bin/EngineConfigCli migrate-project ${projects[@]}
+        else
+          echo "No projects found in ${gitDir}"
+        fi
+
+        mvn -B versions:set-property versions:commit "-Dproperty=project.build.plugin.version" "-DnewVersion=${{ inputs.ivyPluginVersion }}" -DallowSnapshots=true -DprocessAllModules
+        mvn -B versions:set-property versions:commit "-Dproperty=tester.version" "-DnewVersion=${{ inputs.testerVersion }}" -DallowSnapshots=true -DprocessAllModules
```

Comment:
I think setting these versions should no longer be required after migrating projects,
as the ivy-project-parent refers the latest versions of these plugins.